src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.ts

```diff
@@ -110,11 +110,13 @@ export function normalizeClientRequestArgs(
   // and derive request options from it.
   else if (args[0] instanceof URL) {
     url = args[0]
-    logger.info('first argument is a URL:', url)
 
     options = resolveRequestOptions(args, url)
     logger.info('derived request options:', options)
 
+    url.pathname = options.path || ''
```

Comment:
This should be redundant as in this clause the request has been constructed given a `URL` instance instead of the options. You can see above we call `resolveRequestOptions` to derive request options _from_ the given URL. This means that `options.path` should already be derived from `url.pathname` in that function. 

Is there a particular issue that surrounds this? We may start with a test to confirm it.

Reply:
Given this request:
```js
http.get(new url.URL('http://example.test'), { path: '/hello' })
```
The `options.path` is derived correctly, but we may want to populate the `url.pathname` with the correct value because we pass this to the interceptor handler.

> We may start with a test to confirm it.

See the test in the PR.

Reply:
Then I guess it should be 

```ts
url.pathname = options.path || url.pathname
```

We need to check which value takes precedence: the explicit `url.pathname` or `options.path`.

```ts
http.get(new URL('https://example.com/one'), { path: '/two' })
// what would be the actual request URL?
```

Reply:
Great point! 
![image](https://github.com/mswjs/interceptors/assets/11459632/5adef6c1-75ef-4cae-bc79-210cc5115de5)
I guess the `path` option wins

Reply:
Got it. But that fallback value still makes sense otherwise we'll lose the original path set on the URL.

Reply:
I updated the PR 👍